File: day13AbstractClass.cs
using System;
using System.Collections.Generic;
using System.IO;
abstract class Book
{

    protected String title;
    protected  String author;

    public Book(String t,String a){
        title=t;
        author=a;
    }
    public abstract void display();


}
//above has been written
//you only need to be worried about the below:
//Write MyBook class
class MyBook :Book {
    private
     int price;
    public
     MyBook(string title, string author, int price): base(title,author) {
         this.price = price;
     }

    public override void display(){
        Console.WriteLine($"Title: {this.title}");
        Console.WriteLine($"Author: {this.author}");
        Console.WriteLine($"Price: {this.price}");
    }
}

//below has been written
class Solution {
    static void Main(String[] args) {
      String title=Console.ReadLine();
      String author=Console.ReadLine();
      int price=Int32.Parse(Console.ReadLine());
      Book new_novel=new MyBook(title,author,price);
      new_novel.display();
    }
}

File: day13AbstractClass.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
#include <string>
using namespace std;
class Book{
    protected:
        string title;
        string author;
    public:
        Book(string t,string a){
            title=t;
            author=a;
        }
        virtual void display()=0;

};
//above has been given
//only the below class is what we need to write
//Write MyBook class
class MyBook : public Book {
    private:
      int price;
    public:
      MyBook(string title, string author, int price): Book(title,author) {
          this->price = price;
      }
      void display() {
          cout<<"Title: "<<this->title<<endl;
          cout<<"Author: "<<this->author<<endl;
          cout<<"Price: "<<this->price<<endl;
      }
};

//below has been given
int main() {
    string title,author;
    int price;
    getline(cin,title);
    getline(cin,author);
    cin>>price;
    MyBook novel(title,author,price);
    novel.display();
    return 0;
}
